### ISSUE TITLE
stream.push() after EOF

### ISSUE THREAD

**Issue by netmiller** [mscdex/ssh2#233]
I have still some issues playing with examples.
Now I copied second example __Authenticate using keys and start an interactive shell session__ (exactly) and got this :

````
Esan-iMac:iris$node test/test46.js
Client :: ready
events.js:125
      throw er; // Unhandled 'error' event
            ^
Error: stream.push() after EOF
    at readableAddChunk (_stream_readable.js:132:15)
    at SSH2Stream.Readable.push (_stream_readable.js:109:10)
    at SSH2Stream.Transform.push [as __push] (_stream_transform.js:128:32)
    at SSH2Stream.push (/Users/esa/devel/iris/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:217:18)
    at send (/Users/esa/devel/iris/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:4825:12)
    at SSH2Stream.channelClose (/Users/esa/devel/iris/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:785:10)
    at Channel.close (/Users/esa/devel/iris/node_modules/ssh2/lib/Channel.js:283:35)
    at SSH2Stream.<anonymous> (/Users/esa/devel/iris/node_modules/ssh2/lib/Channel.js:107:12)
    at SSH2Stream.g (events.js:240:16)
    at emitNone (events.js:65:13)
````

I don't need interactive session, and I think I can find some workaround, but FYI.
Can it be something with __iojs v.1.20__ ??  My client/workstation is OS X and server to connect is Ubuntu 14.04 Open-SSH server (installed from ubuntu repo).

__Edit:__ forgot to say : package is now ssh2@0.4.4

**Comment by rmg** [comment]
Strange. It works fine for me under node-v0.10.36, node-v0.12.0, iojs-v1.1.0, and iojs-v1.2.0, connecting from a Mac to either localhost or a remote Ubuntu 14.04 host.

**Comment by phawxby** [comment]
I'm experiencing the same error. It usually occurs if I've just put my laptop into standby with an active connection. From a bit of poking around it seems that `Channel.prototype.close` does not correctly detect if the current state is closed due to an untidy disconnect, so it then attempts to perform a tidy close via `this._client._sshstream.channelClose` which then results in attempting to send data to a dead connection, hence the error at stream.push

**Comment by cmp-202** [comment]
If you want a ssh2 shell option try https://www.npmjs.com/package/ssh2shell which is a wrapper class for SSH2 shell command.

**Comment by mscdex** [comment]
Should be fixed in mscdex/ssh2-streams@340048cb74.
